glyphs-reader/src/font.rs

```diff
@@ -2011,6 +2011,41 @@ impl TryFrom<RawShape> for Shape {
     }
 }
 
+// Direct port from
+// <https://github.com/fonttools/fonttools/blob/b7509b2/Lib/fontTools/misc/transform.py#L65-L77>
+const EPSILON: f64 = 1e-15;
+const ONE_EPSILON: f64 = 1.0 - EPSILON;
+const MINUS_ONE_EPSILON: f64 = -1.0 + EPSILON;
+
+fn norm_sin_cos(v: f64) -> f64 {
+    if v.abs() < EPSILON {
+        0.0
+    } else if v > ONE_EPSILON {
+        1.0
+    } else if v < MINUS_ONE_EPSILON {
+        -1.0
+    } else {
+        v
+    }
+}
+
+/// Return [kurbo::Affine] rotation around angle (in degrees) with normalized sin/cos.
+///
+/// This ensures that for the four "cardinal" rotations (0, 90, 180, 270), the values
+/// of sin(angle) and cos(angle) are rounded to *exactly* 0.0, +1.0 or -1.0, thus avoiding
+/// very small values (e.g. 1.2246467991473532e-16) whose effect may be amplified as some
+/// transformed point coordinates end up close to the 0.5 threshold before being rounded
+/// to integer later in the build.
+/// It matches the output of the fontTools' Transform.rotate() used by glyphsLib.
+/// <https://github.com/fonttools/fonttools/blob/b7509b2/Lib/fontTools/misc/transform.py#L246-L258>
+fn normalized_rotation(angle_deg: f64) -> Affine {
```

Comment:
I wonder if any font source actually benefits from the use of epsilon instead of just checking if the value is an exact multiple of 90? - it's so small that it seems unlikely a source has a value that isn't a multiple of 90 and matches. That woudl lead me to implement this as a lookup of a const Affine, perhaps similar to this [sketch](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=e46e9d7288ccf00630d06138306781e3)

Reply:
> I wonder if any font source actually benefits from the use of epsilon

I don't think so, good point! Since we already have angles in degrees, we can just test the 4 cardinal angles directly.

See https://github.com/googlefonts/fontc/pull/1219/commits/f178e899497f453efdc06ab05d3da221d326fee3